gov/governance/_RPC_api_proposal.gno

```diff
@@ -11,21 +12,38 @@ import (
 	en "gno.land/r/gnoswap/v1/emission"
 )
 
+func createProposalJsonNode(id uint64, proposal ProposalInfo) *json.Node {
+	proposalObj := json.ObjectNode("", nil)
+	proposalObj.AppendObject("id", json.StringNode("id", ufmt.Sprintf("%d", id)))
+	proposalObj.AppendObject("configVersion", json.StringNode("configVersion", ufmt.Sprintf("%d", proposal.ConfigVersion)))
+	proposalObj.AppendObject("proposer", json.StringNode("proposer", proposal.Proposer.String()))
+	proposalObj.AppendObject("status", json.StringNode("status", b64Encode(getProposalStatus(id))))
+	proposalObj.AppendObject("type", json.StringNode("type", proposal.ProposalType.String()))
+	proposalObj.AppendObject("title", json.StringNode("title", proposal.Title))
+	proposalObj.AppendObject("description", json.StringNode("description", proposal.Description))
+	proposalObj.AppendObject("vote", json.StringNode("vote", b64Encode(getProposalVotes(id))))
+	proposalObj.AppendObject("extra", json.StringNode("extra", b64Encode(getProposalExtraData(id))))
+	return proposalObj
+}
+
 // GetProposals returns all proposals with necessary information.
 func GetProposals() string {
 	en.MintAndDistributeGns()
 	updateProposalsState()
```

Comment:
Since we're trying to minimize the number of calls to call `en.MintAndDistributeGns()`, it can be removed here? what do you think? @onlyhyde @r3v4s 

Also, `updateProposalsState()` may be an attack vector if it has no call restrictions. Maybe we can add some conditions or remove this?

Reply:
However it changes, apply the same for all functions across the governance contract.